diff --git a/bmi_hydrotrend.c b/bmi_hydrotrend.c
index f8cbe18..91b3ea2 100644
--- a/bmi_hydrotrend.c
+++ b/bmi_hydrotrend.c
@@ -37,9 +37,7 @@ get_component_name (BMI_Model *self, char * name)
 
 
 #define INPUT_VAR_NAME_COUNT (0)
-static const char *input_var_names[INPUT_VAR_NAME_COUNT] = {
-
-};
+static const char **input_var_names = NULL;
 
 
 static int
@@ -460,10 +458,10 @@ get_value_at_indices (BMI_Model *self, const char *name, void *dest,
     { /* Copy the data */
         int i;
         int offset;
-        void * ptr;
+        char * ptr;
         for (i=0, ptr=dest; i<len; i++, ptr+=itemsize) {
             offset = inds[i] * itemsize;
-            memcpy (ptr, src + offset, itemsize);
+            memcpy (ptr, (char*)src + offset, itemsize);
         }
     }
 
@@ -505,10 +503,10 @@ set_value_at_indices (BMI_Model *self, const char *name, int * inds, int len,
     { /* Copy the data */
         int i;
         int offset;
-        void * ptr;
+        char * ptr;
         for (i=0, ptr=src; i<len; i++, ptr+=itemsize) {
             offset = inds[i] * itemsize;
-            memcpy (to + offset, ptr, itemsize);
+            memcpy ((char*)to + offset, ptr, itemsize);
         }
     }
     return BMI_SUCCESS;
